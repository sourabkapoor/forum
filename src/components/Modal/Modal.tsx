import styles from "./Modal.module.css";
import CSSModules from "react-css-modules";

interface Props {
  children: JSX.Element | JSX.Element[];
}

const Modal: React.FC<Props> = ({ children }) => {
  return <div styleName="modal">{children}</div>;
};

export default CSSModules(Modal, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "log",
});
