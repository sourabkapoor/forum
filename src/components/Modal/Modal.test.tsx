import {
  queryByPlaceholderText,
  queryByText,
  render,
  screen,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import Modal from "./Modal";

describe("Modal", () => {
  test("snapshot", () => {
    const { asFragment } = render(
      <Modal>
        <div>Mock Children</div>
      </Modal>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
