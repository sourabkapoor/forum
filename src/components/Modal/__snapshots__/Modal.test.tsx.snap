// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div>
      Mock Children
    </div>
  </div>
</DocumentFragment>
`;
